File: features/Dungeon/AutoUlt.js
import { data } from "../../stuff/guidk"
import { modMessage, useAbility } from "../../utils"

// Auto Ult (for tanks/healers?)
let firstlaser = false
register("worldLoad", () => {
    firstlaser = false
})

register("chat", () => {
    if (!data.auto.options[2]) return
    if (firstlaser) return
    modMessage("§2Frenzy soon... ULT TIME!")
    useAbility()
    firstlaser = true
}).setCriteria("[BOSS] Maxor: YOU TRICKED ME!")

register("chat", () => {
    if (!data.auto.options[2]) return
    if (firstlaser) return
    modMessage("§2Frenzy soon... ULT TIME!")
    useAbility()
    firstlaser = true
}).setCriteria("[BOSS] Maxor: THAT BEAM! IT HURTS! IT HURTS!!")

register("chat", () => {
    if (!data.auto.options[2]) return
    modMessage("§2Goldor time zzz")
    useAbility()
}).setCriteria("[BOSS] Goldor: You have done it, you destroyed the factory…")

register("chat", () => {
    if (!data.auto.options[2]) return
    setTimeout(() => {
        modMessage("§2Giants incoming")
        useAbility()
    }, 1000);
}).setCriteria("[BOSS] Sadan: My giants! Unleashed!")

File: src/main/kotlin/me/odinclient/features/auto/AutoUlt.kt
package me.odinclient.features.auto

import kotlinx.coroutines.DelicateCoroutinesApi
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import me.odinclient.OdinClient
import me.odinclient.utils.ChatUtils
import me.odinclient.utils.PlayerUtils
import net.minecraftforge.client.event.ClientChatReceivedEvent
import net.minecraftforge.event.world.WorldEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

class AutoUlt: OdinClient() {


    private var firstLaser = false
    @SubscribeEvent
    fun onWorldLoad(event: WorldEvent.Load) {
        firstLaser = false
    }

    @SubscribeEvent
    fun ultDetect(event: ClientChatReceivedEvent) {
        when (event.message.unformattedText) {
            "[BOSS] Maxor: YOU TRICKED ME!", "[BOSS] Maxor: THAT BEAM! IT HURTS! IT HURTS!!" -> {
                if (firstLaser) return
                frenzyAndUlt()
                firstLaser = true
            }
            "[BOSS] Goldor: You have done it, you destroyed the factory…" -> {
                ChatUtils.modMessage("§2Goldor time zzz")
                PlayerUtils.dropItem()
            }
            "[BOSS] Sadan: My giants! Unleashed!" -> {
                GlobalScope.launch {
                    delay(1000L)
                    ChatUtils.modMessage("§2Giants incoming")
                    PlayerUtils.dropItem()
                }
            }
        }
    }

    private fun frenzyAndUlt() {
        ChatUtils.modMessage("§2Frenzy soon... ULT TIME!")
        PlayerUtils.dropItem()
    }

}